src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -590,25 +981,53 @@ class module_sram : public module_impl
   void
   sync_if_dirty() override
   {
+#if 0
     if (m_patched_args.size() != m_parent->number_of_arg_patchers()) {
       auto fmt = boost::format("ctrlcode requires %d patched arguments, but only %d are patched")
         % m_parent->number_of_arg_patchers() % m_patched_args.size();
-      throw std::runtime_error{fmt.str()};
+      throw std::runtime_error{ fmt.str() };
     }
+#endif
```

Comment:
We should get rid of this "if 0". It was commented out for the quick prototyping. Need to understand what it is doing and enable it for all use cases.

Reply:
Agree. But this needs to be done in a follow up pull request where we can merge m_arg2patcher (used by TurinAI) with m_arg2patchers (used by IPU)

Reply:
Fixed